lib/features/thread/presentation/thread_controller.dart

```diff
@@ -135,7 +138,12 @@ class ThreadController extends BaseController {
   @override
   void onInit() {
     _onMailboxDashBoardListener();
-    consumeState(_getRecentSearchInteractor.execute());
+    mailboxDashBoardController.searchTyping.listen((query) {
+      if (_searchDebounce?.isActive == false) _searchDebounce?.cancel();
+      _searchDebounce = Timer(const Duration(milliseconds: 500), () {
```

Comment:
Stream support built-in debounce operator. Or you can use Worker of GetX instead.

Reply:
https://pub.dev/documentation/get/latest/get_rx_src_rx_workers_rx_workers/get_rx_src_rx_workers_rx_workers-library.html

Reply:
or: https://github.com/ReactiveX/rxdart/tree/master/lib/src/transformers/backpressure

Reply:
I used rxdart